unicodetools/src/main/java/org/unicode/props/UnicodeProperty.java

```diff
@@ -720,39 +724,90 @@ public static String toSkeleton(String source) {
         if (source == null) return null;
         StringBuffer result = new StringBuffer();
         // remove spaces, medial '-'
         // we can do this with char, since no surrogates are involved
         for (int i = 0; i < source.length(); ++i) {
             char ch = source.charAt(i);
+            final char uppercase = Character.toUpperCase(ch);
+            if (validate && uppercase != ch) {
+                throw new IllegalArgumentException(
+                        "Illegal Name Char: U+" + Utility.hex(ch) + ", " + ch);
+            }
+            ch = uppercase;
             if (('0' <= ch && ch <= '9') || ('A' <= ch && ch <= 'Z') || ch == '<' || ch == '>') {
                 result.append(ch);
             } else if (ch == ' ') {
                 // don't copy ever
             } else if (ch == '-') {
-                // only copy non-medials AND trailing O-E
-                if (0 == i
-                        || i == source.length() - 1
-                        || source.charAt(i - 1) == ' '
-                        || source.charAt(i + 1) == ' '
-                        || (i == source.length() - 2
-                                && source.charAt(i - 1) == 'O'
-                                && source.charAt(i + 1) == 'E')) {
-                    System.out.println("****** EXCEPTION " + source);
+                // Only copy a hyphen-minus if it is non-medial, or if it is
+                // the hyphen in U+1180 HANGUL JUNGSEONG O-E.
+                boolean medial;
+                if (0 == i || i == source.length() - 1) {
+                    medial = false; // Name-initial or name-final.
+                } else {
+                    final char preceding = Character.toUpperCase(source.charAt(i - 1));
+                    final char following = Character.toUpperCase(source.charAt(i + 1));
+                    medial =
+                            (('0' <= preceding && preceding <= '9')
+                                            || ('A' <= preceding && preceding <= 'Z'))
+                                    && (('0' <= following && following <= '9')
+                                            || ('A' <= following && following <= 'Z'));
+                }
+                boolean is1180 = false;
+                if (medial
+                        && i <= source.length() - 2
+                        && Character.toUpperCase(source.charAt(i + 1)) == 'E'
+                        && result.toString().equals("HANGULJUNGSEONGO")) {
+                    is1180 = true;
+                    for (int j = i + 2; j < source.length(); ++j) {
+                        if (source.charAt(j) != ' ' && source.charAt(j) != '_') {
+                            is1180 = false;
+                        }
+                    }
+                }
+                if (!medial || is1180) {
+                    if (validate) {
+                        System.out.println("****** EXCEPTION " + source);
```

Comment:
what is the point of printing this?

Reply:
I am not sure.
This function was only used in TestNameUniqueness, and that is the only place that now calls it with validate=true.

It looks like that test is not actually a test in the sense of the test framework, and the only thing it does is print things out.

… maybe we should have a non-manual test for https://www.unicode.org/policies/stability_policy.html#Name_Uniqueness? 😬

Reply:
> I am not sure. This function was only used in TestNameUniqueness, and that is the only place that now calls it with validate=true.

Let's drop this.

> … maybe we should have a non-manual test for Name_Uniqueness? 😬

We might already have an issue for that.
I don't think it's a high priority because there is a name uniqueness check built into the chart production tooling, so another WG would catch problems.